pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1854,23 +1858,57 @@ private void setRowBufferColumn(Tuple rowBuffer,
         // or we won't be able to re-parse it.
         //
         case Types.DATE:
-          rowBuffer.set(columnIndex, connection
-              .encodeString(
-                  connection.getTimestampUtils().toString(
-                      getDefaultCalendar(), (Date) valueObject)));
+          if (valueObject instanceof Timestamp) {
+            rowBuffer.set(columnIndex, connection
```

Comment:
I am confused by this. We are in the DATE branch, previously we hard casted to `java.sql.Date`, so far so good. But to support `LocalDate` we now test for `java.sql.Timestamp`. Why was this added? Why do we do an `instacenof` check on `java.sql.Timestamp` but then cast to `java.sql.Date`. Wouldn't this always fail?

I believe the `java.sql.Date` path is now broken as this ends up in the `else` branch.

I believe we are missing some unit tests here.

Reply:
I agree, let me build some more unit tests.